checks/license_check.go

```diff
@@ -0,0 +1,146 @@
+// Copyright 2020 Security Scorecard Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package checks
+
+import (
+	"regexp"
+	"strings"
+
+	"github.com/ossf/scorecard/v3/checker"
+	"github.com/ossf/scorecard/v3/checks/fileparser"
+)
+
+type check func(str string, extCheck []string) bool
+
+type checks struct {
+	rstr string // regex string
+	f    check
+	p    []string
+}
+
+// LicenseCheckPolicy is the registered name for LicenseCheck.
+const LicenseCheckPolicy = "LicenseCheck"
+
+//nolint:gochecknoinits
+func init() {
+	registerCheck(LicenseCheckPolicy, LicenseCheck)
+}
+
+const (
+	copying      = "copy(ing|right)"
+	license      = "(un)?licen[sc]e"
+	preferredExt = "*\\.(md|markdown|html)$"
+	anyExt       = ".[^./]"
+	ofl          = "ofl"
+	patents      = "patents"
+)
+
+// regex converted from https://github.com/licensee/licensee/blob/master/lib/licensee/project_files/license_file.rb
```

Comment:
nit: some comments still have low-case letter ;-)

Reply:
Update the comments. Thanks :)